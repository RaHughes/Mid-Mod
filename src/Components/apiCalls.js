
export const getData = () => {
  return fetch('http://localhost:3001/api/v1/reservations')
  .then(res => { 
    return res.json()})
  .catch(err => console.log(err))
}