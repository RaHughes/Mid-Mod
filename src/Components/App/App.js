import React, { Component } from 'react';
import { getData, postData } from '../apiCalls';
import LandingForm from '../LandingForm/LandingForm';
import CardContainer from '../CardContainer/CardContainer';
import './App.css';

class App extends Component {
  constructor() {
  super()
    this.state = {
      reservations: []
  }
}
  componentDidMount() {
    getData()
    .then(reserves => this.setState({ reservations: reserves }))
    .catch(err => console.log(err))
  }

  addReserve = (newObj) => {
    postData(newObj)
    .then(this.setState({ reservations: [...this.state.reservations, newObj] }))
    .catch(err => console.log(err))
  }

  render() {
    return (
      <div className="App">
        <h1 className='app-title'>Turing Cafe Reservations</h1>
        <div className='resy-form'>
          <LandingForm addReserve={this.addReserve}/>
          <CardContainer reservations={this.state.reservations}/>
        </div>
        <div className='resy-container'>
        </div>
      </div>
    )
  }
}

export default App;
