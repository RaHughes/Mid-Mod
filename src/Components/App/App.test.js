import React from 'react';
import { shallow } from 'enzyme';
import { getData, postData } from '../apiCalls';
import App from './App';

jest.mock('../apiCalls.js')

describe('App', () => {
  it('Should match snapshot', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  })

  it('Should call getData', () => {
    getData.mockImplementation(() => {  return Promise.resolve([{id: 1, name: 'Becky', date: '7/30', time: '8:00', number: 2}])})
    shallow(<App />)
    expect(getData).toHaveBeenCalled()
  })

  it('Should add a new Reservation', () => {
    // const mockObj = { id: 2, name: 'Henry Lee Lucas', date: '7/29', time: '7:30', number: 1 }
    postData.mockImplementation(() => { return Promise.resolve({id: 2, name: 'Henry Lee Lucas', date: '7/29', time: '7:30', number: 1})})
    shallow(<App />)
    // addReserve(mockObj)
    expect(postData).toHaveBeenCalled
  })
})

