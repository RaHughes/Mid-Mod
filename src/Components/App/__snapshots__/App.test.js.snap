// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should match snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <div
    className="resy-form"
  >
    <LandingForm
      addReserve={[Function]}
    />
    <CardContainer
      reservations={Array []}
    />
  </div>
  <div
    className="resy-container"
  />
</div>
`;
